File: apps/backend/src/scout_ai_backend/domain/entities/player.py
from dataclasses import dataclass
from enum import StrEnum
from uuid import UUID


class TeamSide(StrEnum):
    HOME = "home"
    AWAY = "away"


@dataclass(slots=True)
class Position:
    x: float
    y: float


@dataclass(slots=True)
class Player:
    id: UUID
    match_id: UUID
    team: TeamSide
    track_id: str
    average_position: Position
    distance_covered_km: float
    heatmap: list[list[float]]
    touches: int
    passes_made: int
    passes_received: int
    rating: float
    name: str | None = None

File: packages/shared/src/types/player.ts
import type { MatchTeamSide } from "./match";

export interface Position {
  x: number;
  y: number;
}

export interface Player {
  id: string;
  team: MatchTeamSide;
  track_id: string;
  name: string | null;
  average_position: Position;
  distance_covered_km: number;
  /** Normalized 0-1 pitch-space grid of visit density. */
  heatmap: number[][];
  touches: number;
  passes_made: number;
  passes_received: number;
  /** Composite score (3-10) from relative distance, touches, and passes — not a subjective rating. */
  rating: number;
}
